src/connector/src/source/cdc/source/reader.rs

```diff
@@ -121,30 +85,27 @@ impl<T: CdcSourceTypeTrait> CommonSplitReader for CdcSplitReader<T> {
             properties.insert("table.name".into(), table_name);
         }
 
+        let source_id = split.split_id() as u64;
+        let source_type = conn_props.get_source_type_pb();
         let (mut tx, mut rx) = mpsc::channel(DEFAULT_CHANNEL_SIZE);
 
         let jvm = JVM
             .get_or_init()
             .map_err(|e| anyhow!("jvm not initialized properly: {:?}", e))?;
 
         let get_event_stream_request = GetEventStreamRequest {
-            source_id: self.source_id,
-            source_type: self.conn_props.get_source_type_pb() as _,
-            start_offset: self.start_offset.unwrap_or_default(),
+            source_id,
+            source_type: source_type as _,
+            start_offset: split.start_offset().clone().unwrap_or_default(),
             properties,
-            snapshot_done: self.snapshot_done,
+            snapshot_done: split.snapshot_done(),
         };
 
-        let source_id = get_event_stream_request.source_id.to_string();
-        let source_type = get_event_stream_request.source_type.to_string();
-
         std::thread::spawn(move || {
```

Comment:
Any special reason for spawning a thread here? How about `spawn_blocking`? cc @chenzl25

Reply:
The thread here acts as the sender which consume events pulled from upstream, and the java side code is a main loop, if we use tokio `spawn_blocking` it will block the tokio runtime thread.

Reply:
I don't think so, as `spawn_blocking` is designed to avoid blocking the runtime. 😕 Either LGTM. I'm just concerned about potential resource leaks.

Reply:
> I don't think so, as spawn_blocking is designed to avoid blocking the runtime
You are right, my mistake. here we don't need to await the result of it, seems no need to spawn_blocking.